src/pipeline.js

```diff
@@ -147,38 +143,41 @@ class Pipeline {
      * @param {String} name - name of the extension point (typically the function name).
      * @param {pipelineFunction} f - a new pipeline step that will be injected relative to `name`.
      */
-    this.attach.before = (name, f) => attachGeneric.bind(this)(name, f, true);
+    this.attach.before = (name, f) => attachGeneric.bind(this)(name, f, -1);
     /**
      * Registers an extension to the pipeline. The function `f` will be run in
      * the pipeline after the function called `name` will be executed. If `name`
      * does not exist, `f` will never be executed.
      * @param {String} name - name of the extension point (typically the function name).
      * @param {pipelineFunction} f - a new pipeline step that will be injected relative to `name`.
      */
-    this.attach.after = (name, f) => attachGeneric.bind(this)(name, f, false);
-  }
-
-  /**
-   * Adds a processing function to the `pre` list to this pipeline.
-   * @param {pipelineFunction} f function to add to the `post` list
-   * @returns {Pipeline} this
-   */
-  before(f) {
-    this.describe(f);
-    this._pres.push(f);
-    this._last = this._pres;
-    return this;
+    this.attach.after = (name, f) => attachGeneric.bind(this)(name, f, 1);
+    /**
+     * Registers an extension to the pipeline. The function `f` will be executed in
+     * the pipeline instead of the function called `name`. If `name` does not exist,
+     * `f` will never be executed.
+     * @param {String} name - name of the extension point (typically the function name).
+     * @param {pipelineFunction} f - a new pipeline step that will replace `name`.
+     */
+    this.attach.replace = (name, f) => attachGeneric.bind(this)(name, f, 0);
   }
 
   /**
-   * Adds a processing function to the `pre` list to this pipeline.
-   * @param {pipelineFunction} f function to add to the `post` list
+   * Adds a processing function to the `step` list of this pipeline.
+   * @param {pipelineFunction} f function to add to the `step` list
    * @returns {Pipeline} this
    */
-  after(f) {
+  use(f) {
     this.describe(f);
-    this._posts.push(f);
-    this._last = this._posts;
+    this._steps.push(f);
+    this._last = this._steps;
+    // check for extensions
+    if (f && (f.before || f.replace || f.after)) {
+      if (typeof this._oncef === 'function') {
+        throw new Error(`Step '${this._oncef.alias}' already registered extensions for this pipeline, refusing to add more with '${f.alias}'.`);
+      }
+      this._oncef = f;
```

Comment:
would rename this to `sealed`....

Reply:
You mean the `_oncef`?

Reply:
or `_cont`? In essence, this is what's left of the continuation function.

Reply:
hmm...I see. maybe rename it to `_overrides` or `_plugins` or `_attachments`

Reply:
but in most cases, it's the `pre` (aka continuation) function...

Reply:
I was wrong, thanks @tripodsan. Renamed it to `_attachments` now.